sympy/functions/elementary/tests/test_complexes.py

```diff
@@ -892,3 +892,18 @@ def test_issue_14238():
     # doesn't cause recursion error
     r = Symbol('r', real=True)
     assert Abs(r + Piecewise((0, r > 0), (1 - r, True)))
+
+def test_zero_assumptions():
+    nr = Symbol('nonreal', real=False)
+    ni = Symbol('nonimaginary', imaginary=False)
+    # imaginary implies not zero
+    nzni = Symbol('nonzerononimaginary', zero=False, imaginary=False)
```

Comment:
maybe a more helpful variable name is `notonimaginaryaxis` but this is just nitpicking.